lib/common/common/src/fixed_length_priority_queue.rs

```diff
@@ -24,7 +24,10 @@ impl<T: Ord> FixedLengthPriorityQueue<T> {
     /// Creates a new queue with the given length
     /// Panics if length is 0
     pub fn new(length: usize) -> Self {
-        let heap = BinaryHeap::with_capacity(length + 1);
+        let mut heap = BinaryHeap::new();
+        heap.try_reserve_exact(length + 1)
+            .expect("Failed to reserve memory for heap");
```

Comment:
This can still crash the whole service under some conditions. 
While following this strategy, the proper solution would require to propagate error to the API via result

Reply:
Right